src/test/java/io/kestra/storage/gcs/GcsStorageTest.java

```diff
@@ -162,4 +170,112 @@ void deleteByPrefixNoResult() throws Exception {
+
+        List<String> path = Arrays.asList(
+            "/" + prefix + "/storage/root.yml",
+            "/" + prefix + "/storage/level1/1.yml",
+            "/" + prefix + "/storage/level1/level2/1.yml",
+            "/" + prefix + "/storage/another/1.yml"
+        );
+        path.forEach(throwConsumer(s -> this.putFile(tenantId, resource, s)));
+
+        boolean deleted = storageInterface.delete(tenantId, new URI("/" + prefix + "/storage/level1"));
+        assertThat(deleted, is(true));
+        assertThat(storageInterface.exists(tenantId, new URI("/" + prefix + "/storage/root.yml")), is(true));
+        assertThat(storageInterface.exists(tenantId, new URI("/" + prefix + "/storage/another/1.yml")), is(true));
+        assertThat(storageInterface.exists(tenantId, new URI("/" + prefix + "/storage/level1/1.yml")), is(false));
+        assertThat(storageInterface.exists(tenantId, new URI("/" + prefix + "/storage/level1/level2/1.yml")), is(false));
+        deleted = storageInterface.delete(tenantId, new URI("/" + prefix + "/storage/root.yml"));
+        assertThat(deleted, is(true));
+        assertThat(storageInterface.exists(tenantId, new URI("/" + prefix + "/storage/root.yml")), is(false));
+    }
+
+    @Test
+    void createDirectory() throws URISyntaxException, IOException {
+        String prefix = IdUtils.create();
+        String tenantId = IdUtils.create();
+
+        storageInterface.createDirectory(tenantId, new URI("/" + prefix + "/storage/level1"));
+        FileAttributes attr = storageInterface.getAttributes(tenantId, new URI("/" + prefix + "/storage/level1"));
+        assertThat(attr.getFileName(), is("level1"));
+        assertThat(attr.getType(), is(FileAttributes.FileType.Directory));
+        assertThat(attr.getSize(), is(0L));
+        assertThat(attr.getLastModifiedTime(), notNullValue());
+    }
+
+    @Test
+    void move() throws Exception {
+        String prefix = IdUtils.create();
+        String tenantId = IdUtils.create();
+        URL resource = GcsStorageTest.class.getClassLoader().getResource("application.yml");
+
+        List<String> path = Arrays.asList(
+            "/" + prefix + "/storage/root.yml",
+            "/" + prefix + "/storage/level1/1.yml",
+            "/" + prefix + "/storage/level1/level2/1.yml",
+            "/" + prefix + "/storage/another/1.yml"
+        );
+        path.forEach(throwConsumer(s -> this.putFile(tenantId, resource, s)));
+
+        storageInterface.move(tenantId, new URI("/" + prefix + "/storage/level1"), new URI("/" + prefix + "/storage/moved"));
+
+        List<FileAttributes> list = storageInterface.list(tenantId, new URI("/" + prefix + "/storage/moved"));
+        assertThat(storageInterface.exists(tenantId, new URI("/" + prefix + "/storage/level1")), is(false));
+        assertThat(list.stream().map(FileAttributes::getFileName).toList(), containsInAnyOrder("level2", "1.yml"));
```

Comment:
Please also list "/" + prefix + "/storage/moved/level2" and ensure it contains 2.yml to make sure subdirectories' content is also moved